packaging/debian/postinst

```diff
@@ -52,20 +51,38 @@ case "$1" in
 	# TODO: should not download binaries if it's only a reconfigure
 	echo "Downloading binaries..."
 	for binary in "storagenode" "storagenode-updater"; do
-	    wget -O /tmp/${binary}.zip ${BINARIES_SERVER}/${binary}_linux_amd64.zip
-	    unzip /tmp/${binary}.zip -d /var/storagenode/
-	    mv /var/storagenode/${binary}_linux_amd64 /var/storagenode/${binary}
+	    tmpfile=$(mktemp /tmp/${binary}.zip.XXXXXX)
+	    wget -O $tmpfile ${BINARIES_SERVER}/${binary}_linux_amd64.zip
+	    unzip $tmpfile -d /var/lib/storagenode
+	    mv /var/lib/storagenode/${binary}_linux_amd64 /var/lib/storagenode/${binary}
 	    # so that we're in PATH easily
-	    ln -s /var/storagenode/${binary} /usr/bin/${binary}
+	    ln -s /var/lib/storagenode/${binary} /usr/bin/${binary}
 	done
-	adduser --system --group --no-create-home --quiet $DAEMON_USER
 
 	if [ "$old_storagedir" != "" ] && [ "$storagedir" != "$old_storagedir" ]; then
 	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
 	else
 	    echo "Storage directory: ${storagedir}"
 	fi
+
+	# storagenode setup
+	/var/lib/storagenode/storagenode setup \
```

Comment:
Is this idempotent? That is, if it gets run after the package is already installed, does it clobber things or otherwise break?

If it is safe to run again, we should probably also execute it for `abort-upgrade`/`abort-remove`/`abort-deconfigure` too, not just `configure`.

If it's not safe, we need a careful check before this to make sure we don't clobber any working things.

Reply:
it will throw if config exists

Reply:
That would cause the script to abort when someone tries to upgrade, and dpkg would see that and try to revert to the old version of the package.

We should probably try to check if config exists first, and skip this step if so.